### PR TITLE
Newsyn

### PR DESCRIPTION
**Pull request by burrbull**
r? @therealprof 

New `syn`, `quote` and `proc_macro2`.

### CODE REVIEW

**Comment by therealprof** [comment]
`Vec<TokenStream>`

Nice cheat! I was actually trying to do the same but getting rid of the `Vec` in the same go.

**Comment by burrbull** [comment]
Please, merge #362 .

File: src/generate/register.rs [hunk 1]
```diff
@@ -213,24 +213,25 @@ pub fn fields(
             } = f.bit_range;
             let sc = f.name.to_sanitized_snake_case();
             let pc = f.name.to_sanitized_upper_case();
-            let pc_r = Ident::from(&*format!("{}_A", pc));
```
Comment (therealprof) [inline src/generate/register.rs]: Why all the `from` -> `new` changes? I've introduced the `from` so the update wouldn't require so many changes.

Comment (burrbull) [reply src/generate/register.rs]: I couldn't to make From on new syn to work.
Maybe some features disabled?

Comment (therealprof) [reply src/generate/register.rs]: Probably. `from` should be the same as `new` but automatically using the current call site as span but maybe that has changed again in a newer version.

Comment (burrbull) [reply src/generate/register.rs]:
```
 the trait `std::convert::From<&str>` is not implemented for `proc_macro2::Ident`
```

Comment (burrbull) [reply src/generate/register.rs]: https://github.com/alexcrichton/proc-macro2/issues/93

Comment (therealprof) [reply src/generate/register.rs]: `proc_macro2::Indent`?

Comment (burrbull) [reply src/generate/register.rs]: syn reexport Ident from proc_macro2 now

Comment (therealprof) [reply src/generate/register.rs]: Meh, this is all so weird.

Comment (therealprof) [reply src/generate/register.rs]: Okay, fine, let's go full on nuts with this. Will need proper testing though.

Comment (therealprof) [reply src/generate/register.rs]: bors try

**Comment by burrbull** [comment]
This branch compiles in configuration with last `syn`:
```toml
quote-next = "1.0.0-rc1"
svd-parser = "0.7"
proc-macro2-next = "1.0.0-rc1"

[dependencies.syn-next]
version = "1.0.0-rc1"
features = ["full","extra-traits"]
```

So, I think PR can be merged.

**Comment by burrbull** [comment]
Rebased on master.

**Comment by burrbull** [comment]
syn v1.0

**Comment by burrbull** [comment]
@therealprof  Is there something blocking merge?

**Comment by burrbull** [comment]
bors retry r=therealprof

**Comment by therealprof** [comment]
I've added you to bors so you should be able to try/retry now on this repository.

**Comment by burrbull** [comment]
Thank you.

**Comment by burrbull** [comment]
bors retry r=therealprof

**Comment by burrbull** [comment]
bors retry r=therealprof

**Pull request merged**